File: Admin_p/Admin_p.Web/Modules/ServerTypes/electroDB/OrderDetailsForm.ts
import { ServiceLookupEditor, IntegerEditor, DecimalEditor, DateEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface OrderDetailsForm {
    OrderId: ServiceLookupEditor;
    ProductId: ServiceLookupEditor;
    Quantity: IntegerEditor;
    UnitPrice: DecimalEditor;
    Subtotal: DecimalEditor;
    CreatedAt: DateEditor;
    UpdatedAt: DateEditor;
}

export class OrderDetailsForm extends PrefixedContext {
    static readonly formKey = 'electroDB.OrderDetails';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!OrderDetailsForm.init)  {
            OrderDetailsForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = IntegerEditor;
            var w2 = DecimalEditor;
            var w3 = DateEditor;

            initFormType(OrderDetailsForm, [
                'OrderId', w0,
                'ProductId', w0,
                'Quantity', w1,
                'UnitPrice', w2,
                'Subtotal', w2,
                'CreatedAt', w3,
                'UpdatedAt', w3
            ]);
        }
    }
}

File: Admin_p/Admin_p.Web/Modules/electroDB/OrderDetails/OrderDetailsForm.cs
using Serenity.ComponentModel;
using System;

namespace Admin_p.electroDB.Forms;

[FormScript("electroDB.OrderDetails")]
[BasedOnRow(typeof(OrderDetailsRow), CheckNames = true)]
public class OrderDetailsForm
{
    public int OrderId { get; set; }
    public int ProductId { get; set; }
    public int Quantity { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal Subtotal { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}